### DISCUSSION TITLE
How to connect Tasmota to MQTT Broker with no Password

### DISCUSSION THREAD

**Discussion by Ianszh** [arendst/Tasmota#16429] [Q&A] [answered]
Hello,

I just recently bought a [ESP32 4 Channel Relay Board (ESP32R4)](https://templates.blakadder.com/robotdyn_ESP32R4.html) with the 8 Relay extension.
I have a RPi running which I has a MQTT broker already running and functioning, but with no User and Password assigned. I've entered the Broker IP in MQTT parameters > Host (192.168.0.03) I've tried to not enter a User and Password, but it's not accepted, I've also tried `MqttPassword 0` and `MqttUser 0` but I can't get it to connect to the broker.

I'm familiar with ESP Easy, but this is the first time I've used Tasmota

Thanks for any help

> 00:00:00.003 HDW: ESP32-D0WD 
00:00:00.026 UFS: FlashFS mounted with 304 kB free
00:00:00.069 CFG: Loaded from File, Count 194
00:00:00.084 QPC: Count 1
00:00:00.197 BRY: Berry initialized, RAM used=3735 bytes
00:00:00.260 Project tasmota - ESP32R4 Version 12.1.1(tasmota)-2_0_4_1(2022-08-25T11:34:17)
00:00:00.998 WIF: Connecting to AP2 ****** Channel 1 BSSId 3C:7A:8A:82:08:70 in mode 11n as ESP32R4...
00:00:02.644 WIF: Connected
00:00:03.208 HTP: Web server active on ESP32R4 with IP address 192.168.0.27
14:50:55.201 MQT: Attempting connection...
14:50:55.257 MQT: Connect failed to 192.168.0.3:1883, rc 5. Retry in 10 sec
14:50:57.934 QPC: Reset
14:51:06.985 MQT: Attempting connection...
14:51:07.019 MQT: Connect failed to 192.168.0.3:1883, rc 5. Retry in 20 sec

**Comment by sfromis** [comment, marked as answer]
My setup is based on the Mosquitto broker configured to allow anonymous connections, which works fine, also with the Tasmota default MQTT user id `DVES_USER` not created in Mosquitto. Hence your symptom of `MQTT_CONNECT_UNAUTHORIZED` indicates that it is your broker not accepting such connections.

**Reply by Ianszh** [reply]
Ok thanks I'll check it later, I too have a Mosquitto broker with ESP Easy devices connecting with no User and Password to the broker and they work fine.

**Reply by barbudor** [reply]
If you have mosquitto, then there should be logs
If not detailed, check the docs for the conf files to have detailed logs

**Reply by sfromis** [reply]
For my setup, I'm having this in `mosquitto.conf`, as Mosquitto version 2 was changed to have more restrictive defaults.
```
listener 1883 0.0.0.0
allow_anonymous true
```

**Reply by Ianszh** [reply]
Yes I set that a while ago, as I couldn't connect after an update

**Reply by Ianszh** [reply]
I've checked the Broker logs (below) and can't see any connection attempt, only my other ESP Easy devices.

I also can't remove the standard user and pass, I thought maybe `MqttPassword 0` and `MqttUser 0` would clear this?

I'm baffled

> Sep 04 15:20:28 DietPi mosquitto[602]: 1662301228: New connection from 192.168.0.7:62489 on port 1883.
Sep 04 15:20:28 DietPi mosquitto[602]: 1662301228: New client connected from 192.168.0.7:62489 as SolarPower_1 (p2, c0, k10).
Sep 04 15:20:44 DietPi mosquitto[602]: 1662301244: Client SolarPower_1 disconnected.
Sep 04 15:20:55 DietPi mosquitto[602]: 1662301255: Saving in-memory database to /var/lib/mosquitto//mosquitto.db.
Sep 04 15:30:04 DietPi mosquitto[602]: 1662301804: New connection from 192.168.0.7:56159 on port 1883.
Sep 04 15:30:04 DietPi mosquitto[602]: 1662301804: New client connected from 192.168.0.7:56159 as SolarPower_1 (p2, c0, k10).
Sep 04 15:30:20 DietPi mosquitto[602]: 1662301820: Client SolarPower_1 disconnected.

<img width="390" alt="Screenshot 2022-09-04 at 16 42 08" src="https://user-images.githubusercontent.com/4294446/188319707-6147d92c-ac08-4a6b-b63f-fc170b3863ea.png">

**Reply by sfromis** [reply]
I'd not expect it to help any bit to try to "clear" MqttUser. My experience is that I can connect just fine with the default user/password even without any such configured in Mosquitto. Nevertheless, I'm still doing `MqttUser 0` and `MqttPassword 0` in my scripted standard Tasmota device enrollment setup, and I do no longer see the default `DVES_USER` in the web UI config page for MQTT.

**Reply by barbudor** [reply]
I just started a mosquitto 2.0.15 with config file:
```
allow_anonymous true
listener 1883 0.0.0.0
log_type error
log_type warning
log_type notice
log_type information
```

A Tasmota 12.1.1.1
Right after `backlog mqttuser 0; mqttpassword 0; restart 1`, device restart and connect smoothly to the broker
```
17:50:26.069 MQT: Connected
17:50:26.072 MQT: tele/dev-4119/LWT = Online (retained)
17:50:26.074 MQT: cmnd/dev-4119/POWER = 
17:50:26.081 MQT: tele/dev-4119/INFO1 = {"Info1":{"Module":"Generic","Version":"12.1.1.1(bhome)","FallbackTopic":"cmnd/3C71BF2AD017_fb/","GroupTopic":"cmnd/alldevices/"}}
17:50:26.089 MQT: tele/dev-4119/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"dev-4119","IPAddress":"192.168.168.199"}}
17:50:26.100 MQT: tele/dev-4119/INFO3 = {"Info3":{"RestartReason":"Software/System restart","BootCount":6}}
17:50:29.594 CMD: backlog mqttuser ; mqttpassword
17:50:29.669 MQT: stat/dev-4119/MQTTUSER = {"MqttUser":""}
17:50:29.877 MQT: stat/dev-4119/MQTTPASSWORD = {"MqttPassword":"****"}
```

And mosquitto log:
```
1662306625: New connection from 192.168.168.199:55994 on port 1883.
1662306625: New client connected from 192.168.168.199:55994 as 3C71BF2AD017 (p2, c1, k30).
```

Where `3C71BF2AD017 ` is my MqttClient (%012X)

**Reply by barbudor** [reply]
Just tried the same with MqttUser as DUMMY_USER and MqttPassword DUMMY_PASSWORD and it didn't make a change : connected directly

@lanszh I see that your MqttClient is set to ESP32R4
Do you have any other device using that client name ?
I remind you that MqttClient MUST be unique across all the system. Ech device must have its own MqttClient (including Tasmotas, ESPeasy, PCs, ...)

**Reply by Ianszh** [reply]
Yesssssss `backlog mqttuser 0; mqttpassword 0; restart 1` worked, now we're talking,


> Sep 05 16:46:30 DietPi mosquitto[602]: 1662392790: New connection from 192.168.0.27:63186 on port 1883.
Sep 05 16:46:30 DietPi mosquitto[602]: 1662392790: New client connected from 192.168.0.27:63186 as ESP32R4 (p2, c1, k30).

Yes I'm aware to have unique naming.

Another question if I may?

I'm going to be sending MQTT commands to this with Node-Red, how do I setup Domoticz to accept MQTT Topics? On this [page](https://tasmota.github.io/docs/Domoticz/#virtual-switch) the screenshot shows "In Topic domoticz/in - Out Topic domoticz/out" but my Domoticz doesn't have this!

<img width="409" alt="Screenshot 2022-09-05 at 19 58 46" src="https://user-images.githubusercontent.com/4294446/188499350-fdf03853-8fd8-4b89-8e0b-b1d2011b58d7.png">

**Reply by barbudor** [reply]
Not sure to understand the relation between Tasmota, node-red and domoticz

Domoticz expect data on specific topics which is why there are some configurations to do I order for Tasmota to talk to Domoticz.
Not using Domoticz myself, I can't help you much on that 

Node red is fully flexible and can subscribe/publish to any topic scheme whether tasmota or Domoticz.

**Reply by sfromis** [reply]
Not using Domoticz either, but my understanding is that you do not set it up to accept your MQTT topics, the standard structure prescribes which topics it is already subscribing to, and Tasmota has features to adapt to that.
